org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -55,4 +63,58 @@ protected void addEdits(IDocument document, TextEdit editRoot) throws CoreExcept
 
 		resultingCollections.add(proposal);
 	}
+
+	private static void getGetterSetterProposal(IInvocationContext context, List<CUCorrectionProposal> resultingCollections) {
+		final IType type = getSelectionType(context);
+		try {
+			if (!GenerateGetterSetterOperation.supportsGetterSetter(type)) {
+				return;
+			}
+		} catch (JavaModelException e) {
+			return;
+		}
+
+		ICompilationUnit unit = context.getCompilationUnit();
+		CUCorrectionProposal proposal = new CUCorrectionProposal(ActionMessages.GenerateGetterSetterAction_label, CodeActionKind.Source, unit, null, IProposalRelevance.GENERATE_GETTER_AND_SETTER) {

```

Comment:
Create a constant for jdt.ls specific kinds.
SOURCE_ACTION_GENERATE_KIND = CodeActionKind.Source+".generate";
SOURCE_ACTION_GENERATE_ACCESSORS_KIND = SOURCE_ACTION_GENERATE_KIND+".accessors";

Use SOURCE_ACTION_GENERATE_ACCESSORS_KIND

Reply:
Kinds are hierarchical, so you need to also change `ClientPreferences#isSupportedCodeActionKind` so that it accepts sub kinds : 

```java
	public boolean isSupportedCodeActionKind(String kind) {
		//@formatter:off
		return v3supported && capabilities.getTextDocument().getCodeAction() != null
				&& capabilities.getTextDocument().getCodeAction().getCodeActionLiteralSupport() != null
				&& capabilities.getTextDocument().getCodeAction().getCodeActionLiteralSupport().getCodeActionKind() != null
				&& capabilities.getTextDocument().getCodeAction().getCodeActionLiteralSupport().getCodeActionKind().getValueSet() != null
				&& capabilities.getTextDocument().getCodeAction().getCodeActionLiteralSupport().getCodeActionKind().getValueSet()
				.stream().filter(k -> kind.startsWith(k)).findAny().isPresent();
		//@formatter:on
	}
```